src/lib/expression/placeholder_expression.cpp

```diff
@@ -0,0 +1,47 @@
+#include "placeholder_expression.hpp"
+
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+#include "boost/functional/hash.hpp"
+
+#include "resolve_type.hpp"
+
+namespace opossum {
+
+PlaceholderExpression::PlaceholderExpression(const ParameterID parameter_id)
+    : AbstractParameterExpression(ParameterExpressionType::Placeholder, parameter_id) {}
+
+std::shared_ptr<AbstractExpression> PlaceholderExpression::deep_copy() const {
+  return std::make_shared<PlaceholderExpression>(parameter_id);
+}
+
+std::string PlaceholderExpression::as_column_name() const {
+  std::stringstream stream;
+  stream << "Placeholder[id=" << std::to_string(parameter_id) << "]";
+  return stream.str();
+}
+
+DataType PlaceholderExpression::data_type() const { Fail("Cannot obtain DataType of placeholder"); }
+
+bool PlaceholderExpression::is_nullable() const {
+  // Assuming it's nullable is safe
```

Comment:
This comment doesn't help anyone. Why is this the case?

Reply:
Because no one really listens to it?

Reply:
Seriously though, I replaced it with a Fail(), you're not supposed to query the nullability of placeholders